jmclient/jmclient/wallet.py

```diff
@@ -1013,6 +979,381 @@ def close(self):
     def __del__(self):
         self.close()
 
+class DummyKeyStore(btc.KeyStore):
+    @classmethod
+    def from_iterable(cls, iterable, **kwargs):
+        kstore = cls(**kwargs)
+        for k in iterable:
+            kstore.add_key(k)
+        return kstore
+
+    def add_key(self, k):
+        if isinstance(k, btc.CKeyBase):
+            if k.pub.key_id in self._privkeys:
+                assert self._privkeys[k.pub.key_id] == k
+            else:
+                self._privkeys[k.pub.key_id] = k
+        else:
+            raise ValueError('object supplied to add_key is of unrecognized type')
+
+class PSBTWalletMixin(object):
+    """
+    Mixin for BaseWallet to provide BIP174
+    functions.
+    """
+    def __init__(self, storage, **kwargs):
+        super(PSBTWalletMixin, self).__init__(storage, **kwargs)
```

Comment:
Note that for non-`@classmethod`s, just `super()` is usually enough to call a method of superclass in python3

Reply:
Thanks for the tip, was not aware. Was going to change this then realised it's all over the codebase, so better addressed in a separate commit, will open an issue so as not to forget. Thanks.

Reply:
See https://github.com/JoinMarket-Org/joinmarket-clientserver/issues/566